WikipediaUnitTests/Code/WMFBlacklistTests.m

```diff
@@ -0,0 +1,53 @@
+//
+//  WMFBlacklistTests.m
+//  Wikipedia
+//
+//  Created by Corey Floyd on 1/21/16.
+//  Copyright © 2016 Wikimedia Foundation. All rights reserved.
+//
+
+#import <XCTest/XCTest.h>
+#import "WMFAsyncTestCase.h"
+#import "WMFRelatedSectionBlackList.h"
+#import "MWKTitle.h"
+
+@interface WMFRelatedSectionBlackList (WMFTesting)
+
++ (instancetype)loadFromDisk;
+
+@end
+
+
+@interface WMFBlacklistTests : WMFAsyncTestCase
+
+@end
+
+@implementation WMFBlacklistTests
+
+- (void)tearDown {
+    WMFRelatedSectionBlackList* bl = [[WMFRelatedSectionBlackList alloc] init];
+    [bl removeAllEntries];
+    [bl save];
+    [super tearDown];
+}
+
+- (void)testPersistsToDisk {
+    PushExpectation();
+    MWKTitle* title                = [[MWKTitle alloc] initWithString:@"some-title" site:[MWKSite siteWithCurrentLocale]];
+    WMFRelatedSectionBlackList* bl = [[WMFRelatedSectionBlackList alloc] init];
+    [bl addEntry:title];
+    [bl save].then(^(){
+        [self popExpectationAfter:nil];
+    }).catch(^(NSError* error){
+        XCTFail(@"Error callback erroneously called with error %@", error);
+    });
+    WaitForExpectations();
+
+    bl = [WMFRelatedSectionBlackList loadFromDisk];
+    MWKTitle* first = [[bl entries] firstObject];
+
+    XCTAssertTrue([title isEqual:first],
```

Comment:
> without ruining code completion and showing errors (since you have to use @import to get it to work)

I've run into this issue as well. I filed a radar for it, but the workaround is to disable "Gather code coverage" in the scheme's "Test" action options and then build for testing.  this should regenerate the missing indexes and remove the errors (which are actually indexing errors, not compilation).